### PR TITLE
Sign mode eip191

### PR DESCRIPTION
**Pull request by sunnya97**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by sunnya97** [comment]
We should consider converting this to use YAML instead of JSON.  As long as there's a deterministic mapping from json to yaml

Maybe using something like this:  https://github.com/ghodss/yaml

**Comment by ValarDragon** [comment]
TODO Here: Prettify the JSON in GetSignBytes / w/e its called.

**Comment by sunnya97** [comment]
Decided to do prettified JSON instead of YAML

File: x/auth/tx/eip191_legacy_json.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+func (s signModeEIP191LegacyJSONHandler) DefaultMode() signingtypes.SignMode {
+	return signingtypes.SignMode_SIGN_MODE_EIP191_LEGACY_JSON
+}
+
+func (s signModeEIP191LegacyJSONHandler) Modes() []signingtypes.SignMode {
+	return []signingtypes.SignMode{signingtypes.SignMode_SIGN_MODE_EIP191_LEGACY_JSON}
+}
+
+func (s signModeEIP191LegacyJSONHandler) GetSignBytes(mode signingtypes.SignMode, data signing.SignerData, tx sdk.Tx) ([]byte, error) {
```
Comment (ValarDragon) [inline x/auth/tx/eip191_legacy_json.go:34]: Change this to use prettified JSON

File: x/auth/tx/eip191_legacy_json.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	body := protoTx.tx.Body
+
+	if len(body.ExtensionOptions) != 0 || len(body.NonCriticalExtensionOptions) != 0 {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "SignMode_SIGN_MODE_EIP191_LEGACY_JSON does not support protobuf extension options.")
+	}
+
+	aminoJSONBz := legacytx.StdSignBytes(
+		data.ChainID, data.AccountNumber, data.Sequence, protoTx.GetTimeoutHeight(),
+		legacytx.StdFee{Amount: protoTx.GetFee(), Gas: protoTx.GetGas()},
+		tx.GetMsgs(), protoTx.GetMemo(),
```
Comment (ValarDragon) [inline x/auth/tx/eip191_legacy_json.go:57]: I think prettified JSON happens here

**Comment by tac0turtle** [comment]
Hey, would love to see this upstreamed, let me know what is needed from the sdk for this

**Pull request closed without merging**